src/ape/managers/query.py

```diff
@@ -111,9 +112,10 @@ def query(self, query: QueryType, engine_to_use: Optional[str] = None) -> Iterat
 
         # Go fetch the result from the engine
         result = engine.perform_query(query)
+        data = [val for val in result]
```

Comment:
Without doing this, when we hand `engine.update_cache` the map, the generator forgets, and we lose the return at the end